File: sprint6/By lessons/lesson6code1/lesson6code1.js
let order = [];
let color = new Array(n).fill(0);

function topSort(v) {
    color[v] = 1;
    for (let w of adjList[v]) {
        if (color[w] == 0) {
            topSort(w);
        }
    }
    color[v] = 2;
    order.push(v);
}

function mainTopSort() {
    for (let i = 0; i < n; i++) {
        if (color[i] == 0) {
            topSort(i);
        }
    }
}

File: sprint6/By lessons/lesson6code1/lesson6code1.swift
import Foundation

enum Color {
    case white
    case gray
    case black
}

var order: [Int] = []
var color: [Color] = Array(repeating: .white, count: n)  // Замените n на нужное значение

func topSort(_ v: Int, adjList: [[Int]]) {
    color[v] = .gray
    for w in adjList[v] {
        if color[w] == .white {
            topSort(w, adjList: adjList)
        }
    }
    color[v] = .black
    order.append(v)
}

func mainTopSort(n: Int, adjList: [[Int]]) {
    for i in 0..<n {
        if color[i] == .white {
            topSort(i, adjList: adjList)
        }
    }
}
